src/Flysystem/FlysystemAssetStore.php

```diff
@@ -318,6 +317,53 @@ public function delete($filename, $hash)
         return $protected || $public;
     }
 
+    public function rename($filename, $hash, $newName)
+    {
+        return $this->copyOrRename($filename, $hash, $newName, true);
+    }
+
+    public function copy($filename, $hash, $newName)
+    {
+        return $this->copyOrRename($filename, $hash, $newName, false);
+    }
+
+    /**
+     * Copy or move a file and all variants
+     *
+     * @param string $filename urrent filename
+     * @param string $hash Hash
+     * @param string $newName filename
+     * @param bool $asRename Set to true to rename (instead of copy)
```

Comment:
Yeah but private would make a rather complicated behaviour un-overridable. I'd rather duplicate the method inline and switch a few method calls instead.